frontend/src/components/custom-tools/prompt-card/PromptCardItems.jsx

```diff
@@ -61,18 +100,127 @@ function PromptCardItems({
+  const handleException = useExceptionHandler();
+  const { setAlertDetails } = useAlertStore();
+  const componentWidth = windowWidth * 0.4;
 
-  useEffect(() => {
-    setExpandCard(true);
-  }, [isSinglePassExtractLoading]);
+  const divRef = useRef(null);
 
   const enableEdit = (event) => {
     event.stopPropagation();
     setExpandCard(true);
     setIsEditingTitle(true);
     setIsEditingPrompt(true);
   };
+  const getModelOrAdapterId = (profile, adapters) => {
+    const result = { conf: {} };
+    const keys = ["vector_store", "embedding_model", "llm", "x2text"];
+
+    keys.forEach((key) => {
+      const adapterName = profile[key];
+      const adapter = adapters?.find(
+        (adapter) => adapter?.adapter_name === adapterName
+      );
+      if (adapter) {
+        result.conf[key] = adapter?.model || adapter?.adapter_id?.split("|")[0];
+        if (adapter?.adapter_type === "LLM") result.icon = adapter?.icon;
+      }
+    });
+    return result;
+  };
+
+  const getAdapterInfo = async () => {
+    privateAxios
+      .get(`/api/v1/unstract/${sessionDetails?.orgId}/adapter/`)
+      .then((res) => {
+        const adapterData = res?.data;
+
+        // Update llmProfiles with additional fields
+        const updatedProfiles = llmProfiles?.map((profile) => {
+          return { ...getModelOrAdapterId(profile, adapterData), ...profile };
+        });
+        setLlmProfileDetails(
+          updatedProfiles
+            .map((profile) => ({
+              ...profile,
+              isDefault: profile?.profile_id === selectedLlmProfileId,
+              isEnabled: enabledProfiles.includes(profile?.profile_id),
+            }))
+            .sort((a, b) => {
+              if (a?.isDefault) return -1; // Default profile comes first
+              if (b?.isDefault) return 1;
+              if (a?.isEnabled && !b?.isEnabled) return -1; // Enabled profiles come before disabled
+              if (!a?.isEnabled && b?.isEnabled) return 1;
+              return 0;
+            })
+        );
+      })
+      .catch((err) => {
+        setAlertDetails(handleException(err));
+      });
+  };
 
+  const tooltipContent = (adapterConf) => (
+    <div>
+      {Object.entries(adapterConf)?.map(([key, value]) => (
+        <div key={key}>
+          <strong>{key}:</strong> {value}
+        </div>
+      ))}
+    </div>
+  );
+
+  const handleExpandClick = (profile) => {
+    const profileId = profile?.profile_id;
+    setExpandedProfiles((prevState) =>
+      prevState.includes(profileId)
+        ? prevState.filter((id) => id !== profileId)
+        : [...prevState, profileId]
+    );
+  };
+
+  const handleTagChange = (checked, profileId) => {
+    setEnabledProfiles((prevState) =>
+      checked
+        ? [...prevState, profileId]
+        : prevState.filter((id) => id !== profileId)
+    );
+  };
+
+  const getColSpan = () => {
+    if (componentWidth < 1200) {
+      return 24;
+    } else {
+      return 6;
+    }
+  };
```

Comment:
NIT: Can we simplify this function like this?

```suggestion
  const getColSpan = () => componentWidth < 1200 ? 24 : 6;
```

Reply:
Yes